openlibrary/utils/lcc.py

```diff
@@ -0,0 +1,185 @@
+author last name, city, or whatever. Each character maps to a letter range, so for
+example:
+  - Idaho -> I33 -> I[d][a-d]
+  - Campbell -> C36 -> C[a][m-o]
+
+For the full mapping of character to letter ranges, see [1] Appendix B1 (page 355).
+
+Because the number part of a cutter number maps to letters, even the numeral is sorted
+lexicographically, so for example this is the correct sorting:
+    [I2, I23, I3], **not** [I2, I3, I23]
+
+In essence they are sort of sorted as if they were decimal numbers.
+
+### 3. Specification
+
+These aren't very well defined and could be just about anything. They usually include at
+least the publication year of the edition, but might include edition numbers.
+
+## Sorting
+
+To get _fully_ sortable LCCs, you likely need to use a multipart scheme (as described in
+[2]). That's not really feasible for our Solr instance (especially since we store info
+at the work level, which likely has multiple LCCs). The added complexity of that
+approach is also not immediately worth it right now (but might be in the future).
+
+As a compromise, we make the class number and the first cutter sortable by making the
+class number fixed-width. For example:
+ - PZ73.S758345255 2011 -> PZ-0073.00000000.S758345255 2011
+ - PZ8.3.G276Lo 1971    -> PZ-0008.30000000.G276Lo 1971
+
+This allows for range queries that could include the first cutter. It sorts incorrectly
+if:
+  - The decimal of the class number is longer than 8 digits (few such cases in OL db)
+  - The sort is determined by information that appears after the first cutter
+  - The first cutter is a "double cutter", e.g. B945.D4B65 199
+
+But it works for subject-related range queries, so we consider it sufficient.
+
+## Further Reading
+
+- Wagner, Scott etal. "A Comprehensive Approach to Algorithmic Machine Sorting of
+    Library of Congress Call Numbers" (2019) [1]
+- LCTS/CCS-PCC Task Force on  Library of Congress Classification Training. "Fundamentals
+    of Library of Congress Classification" (????) [2]
+- LCC subjects as PDFs https://www.loc.gov/catdir/cpso/lcco/
+- LCC subjects "walkable" tree http://id.loc.gov/authorities/classification.html
+
+## References
+
+[1]: https://www.terkko.helsinki.fi/files/9666/classify_trnee_manual.pdf
+[2]: https://ejournals.bc.edu/index.php/ital/article/download/11585/9839/
+"""
+import re
+
+from openlibrary.utils.ddc import collapse_multiple_space
+
+LCC_PARTS_RE = re.compile(r'''
+    ^
+    # trailing dash only valid in "sortable" LCCs
+    # Include W, even though technically part of NLM system
+    (?P<letters>[A-HJ-NP-VWZ][A-Z-]{0,2})
+    \s?
+    (?P<number>\d{1,4}(\.\d+)?)?
+    (?P<cutter1>[\s.][^\d\s\[]{1,3}\d*\S*)?
+    (?P<rest>\s.*)?
+    $
+''', re.IGNORECASE | re.X)
+
+
+def short_lcc_to_sortable_lcc(lcc):
+    """
+    See Sorting section of doc above
+    :param str lcc: unformatted lcc
+    :rtype: basestring|None
+    """
```

Comment:
```suggestion
    """
    if lcc == '*':
        return lcc
```

Reply:
✖ See other comment; opting out since `*` is only valid in the context of a range.